src/components/edit-movie/edit-movie.tsx

```diff
@@ -23,140 +21,194 @@ interface Props {
   action: Actions;
 }
 
+interface Values {
+  title: string;
+  release_date: string;
+  poster_path: string;
+  vote_average: number;
+  genres: string[];
+  runtime: number;
+  overview: string;
+}
+
 export const EditMovie = ({ title, movie, onClose, action }: Props) => {
   const dispatch = useAppDispatch();
 
-  const [movieData, setMovieData] = useState<MovieFormValues | undefined>(
-    movie,
-  );
-
-  const handleSubmit = () => {
-    if (!movieData || !isValidMovie(movieData)) {
+  const onSubmit = (values: Values) => {
+    if (!values || !isValidMovie(values)) {
       return;
     }
 
     if (action === Actions.Add) {
-      dispatch(addMovieAction(movieData));
+      dispatch(addMovieAction(values));
     }
     if (movie?.id) {
-      dispatch(updateMovieAction({ ...movieData, id: movie.id }));
+      dispatch(updateMovieAction({ ...values, id: movie.id }));
     }
     onClose();
   };
   const handleReset = () => {
     onClose();
   };
+  const required = (value: any) => (value ? undefined : 'Required');
```

Comment:
1. `any` is an evil, let's use `unknown`.
2. Not really agree with name of function. From it's name it seems to be that it returns boolean value, but in reality it returns `string`. So, from the official docs example they are calling it `getValidator`. I suppose we can do the same :)